python/python/tests/test_scan.py

```diff
@@ -251,6 +313,70 @@ async def test_cluster_scan_all_types(self, glide_client: GlideClusterClient):
         assert not set(encoded_list_keys).intersection(set(keys))
         assert not set(encoded_zset_keys).intersection(set(keys))
 
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_cluster_scan_non_covered_slots(
+        self,
+        protocol: ProtocolVersion,
+        function_scoped_cluster: ValkeyCluster,
+        glide_client_scoped: GlideClusterClient,
+    ):
+        key = get_random_string(10)
+        for i in range(10000):
+            try:
+                await glide_client_scoped.set(f"{key}:{i}", "value")
+            except RequestError as e:
+                continue
+        cursor = ClusterScanCursor()
+        result = await glide_client_scoped.scan(cursor)
+        cursor = cast(ClusterScanCursor, result[0])
+        assert not cursor.is_finished()
+        await glide_client_scoped.config_set({"cluster-require-full-coverage": "no"})
+        # forget one server
+        address_to_forget = glide_client_scoped.config.addresses[0]
+        all_other_addresses = glide_client_scoped.config.addresses[1:]
+        id_to_forget = await glide_client_scoped.custom_command(
+            ["CLUSTER", "MYID"],
+            ByAddressRoute(address_to_forget.host, address_to_forget.port),
+        )
+        for address in all_other_addresses:
```

Comment:
is there a reason nor to perform AllNodes() ?

Reply:
I need all of them to forget before gossip, otherwise it will be flaky